deployment/chart/values.yaml

```diff
@@ -24,5 +24,9 @@ config:
     - harbor.toolforge.org
     - harbor.toolsbeta.wmflabs.org
 
+  # see https://wikitech.wikimedia.org/wiki/Wikimedia_Cloud_Services_team/EnhancementProposals/Decision_record_T330102_What_buildpacks_to_allow_and_include_for_toolforge_build_service_beta
+  builders:
+    - harbor.toolsbeta.wmflabs.org/toolforge/heroku-builder-classic:22
```

Comment:
Is there any reason to choose `heroku-builder-classic` instead of `heroku-builder`? The [official docs](https://github.com/heroku/builder) seem to suggest using the latter when possible.

Reply:
PHP support is indeed a good reason, but if I'm reading the docs correctly, they both implement the CNB spec and the `-classic` ones are the ones that *"don't take advantage of many CNB features"*. So the non-classic ones seem to be the most CNB-oriented.

Reply:
Yep, it's not clear to me, but the `-classic` ones say:
> builder images feature Heroku's classic platform buildpacks, shimmed for compatibility with the Cloud Native Buildpacks specification.

while the others do not, so I was guessing that means the the others are not compatible with the "Cloud Native Buildpacks specification"

Reply:
That readme is super confusing and I had to read it 4 times 😆 

My understanding is that `heroku/builder` are newer packages, natively designed to follow the CNB specification (*native Cloud Native Buildpacks*), while `heroku/builder-classic` are the old Heroku buildpacks (designed before CNB) that were adapted (*shimmed*) to be compatible with the CNB specification.

The readme also says that `heroku/builder` are "suggested", but then the [Python buildpack based on that builder](https://github.com/heroku/buildpacks-python) is marked as "experimental".

Maybe we can start with the `builder-classic` ones and consider migrating to the newer ones later if/when they add PHP support and remove the "experimental" flag.

Reply:
I did a test locally, and I was able to build and run wm-lol (the example dummy app, uwsgi + flask) with the builder-classic, but not just with the builder:

```
#### With builder
dcaro@vulcanus$ docker run --rm -ti --publish 8000 wm-lol
[uWSGI] getting INI configuration from uwsgi.ini
*** Starting uWSGI 2.0.21 (64bit) on [Tue Mar 14 09:30:09 2023] ***
compiled with version: 11.3.0 on 01 January 1980 00:00:01
os: Linux-6.1.0-2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.7-1 (2023-01-18)
nodename: 7c8c922224c5
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 8
current working directory: /workspace
detected binary path: /layers/heroku_python/dependencies/bin/uwsgi
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address :8000 fd 3
Python version: 3.11.2 (main, Feb  8 2023, 12:54:20) [GCC 11.3.0]
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = (not set)
  program name = '/layers/heroku_python/dependencies/bin/uwsgi'
  isolated = 0
  environment = 1
  user site = 1
  safe_path = 0
  import site = 1
  is in build tree = 0
  stdlib dir = '/app/.heroku/python/lib/python3.11'
  sys._base_executable = '/layers/heroku_python/dependencies/bin/uwsgi'
  sys.base_prefix = '/app/.heroku/python'
  sys.base_exec_prefix = '/app/.heroku/python'
  sys.platlibdir = 'lib'
  sys.executable = '/layers/heroku_python/dependencies/bin/uwsgi'
  sys.prefix = '/app/.heroku/python'
  sys.exec_prefix = '/app/.heroku/python'
  sys.path = [
    '/app/.heroku/python/lib/python311.zip',
    '/app/.heroku/python/lib/python3.11',
    '/app/.heroku/python/lib/python3.11/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007fecea4f1440 (most recent call first):
  <no Python frame>

#### with builder-classic
dcaro@vulcanus$ docker run --rm -ti --publish 8001:8000 wm-lol
[uWSGI] getting INI configuration from uwsgi.ini
*** Starting uWSGI 2.0.21 (64bit) on [Tue Mar 14 09:56:26 2023] ***
compiled with version: 11.3.0 on 14 March 2023 09:55:15
os: Linux-6.1.0-2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.7-1 (2023-01-18)
nodename: d904c3255c35
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 8
current working directory: /workspace
detected binary path: /workspace/.heroku/python/bin/uwsgi
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address :8000 fd 3
Python version: 3.11.2 (main, Feb  8 2023, 12:54:20) [GCC 11.3.0]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x7f52663be5b8
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 364600 bytes (356 KB) for 4 cores
*** Operational MODE: preforking ***
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x7f52663be5b8 pid: 1 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 1)
spawned uWSGI worker 1 (pid: 13, cores: 1)
spawned uWSGI worker 2 (pid: 14, cores: 1)
spawned uWSGI worker 3 (pid: 15, cores: 1)
spawned uWSGI worker 4 (pid: 16, cores: 1)
{address space usage: 86843392 bytes/82MB} {rss usage: 29687808 bytes/28MB} [pid: 13|app: 0|req: 1/1] 172.17.0.1 () {24 vars in 255 bytes} [Tue Mar 14 09:56:59 2023] GET / => generated 5684 bytes in 31 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches o
n core 0)
```


I did not try to debug further the `builder` one though, but it seems to be missing some stuff in the path (or maybe it expects you to inject that, something that might happen inside the heroku platform)

Reply:
Actually, I was able to run it if I set two env vars :/

```
dcaro@vulcanus$ docker run --rm --env LD_LIBRARY_PATH=/layers/heroku_python/python/lib/ --env PYTHONHOME=/layers/heroku_python/python/ --publish 8000:8000 wm-lol
```

Not nice though

Reply:
I couldn't find docs on how to use the new buildpacks, maybe that's why they are marked as "experimental"...  It seems safer to go with the "classic" ones for now.

Reply:
Ack, let's see with -classic, and change to the other if we find errors.